helix-term/src/ui/mod.rs

```diff
@@ -255,6 +256,54 @@ pub fn file_picker(root: PathBuf, config: &helix_view::editor::Config) -> Picker
     picker
 }
 
+pub fn changed_file_picker(editor: &helix_view::Editor) -> Picker<FileChange> {
+    let cwd = std::env::current_dir().unwrap_or_else(|_| PathBuf::from("./"));
+
+    let added = editor.theme.get("diff.plus");
+    let deleted = editor.theme.get("diff.minus");
+    let modified = editor.theme.get("diff.delta");
+
+    let picker = Picker::new(
+        Vec::new(),
+        menu::FileChangeData {
+            cwd: cwd.clone(),
+            style_untracked: added,
+            style_modified: modified,
+            style_deleted: deleted,
+            style_renamed: modified,
+        },
+        |cx, meta: &FileChange, action| {
+            let path_to_open = meta.path();
+            if let Err(e) = cx.editor.open(path_to_open, action) {
+                let err = if let Some(err) = e.source() {
+                    format!("{}", err)
+                } else {
+                    format!("unable to open \"{}\"", path_to_open.display())
+                };
+                cx.editor.set_error(err);
+            }
+        },
+    )
+    .with_preview(|_editor, meta| Some((meta.path().to_path_buf().into(), None)));
+    let injector = picker.injector();
+
+    let diff_providers = editor.diff_providers.clone();
+
+    std::thread::spawn(move || {
+        // There's no way we can notify the editor since we're in a background thread. This just
+        // silently fails.
+        if let Ok(change_iter) = diff_providers.get_changed_files(&cwd) {
+            // We ignore errors as they could be caused by permission issues etc
```

Comment:
You can use `report_blocking` to send an error to the main loop for display. I do Thi k reporting the error would be useful.

Reply:
I didn't know this exists. I will update the code to surface both initialization error and iteration errors.